bucketeer2/src/test/kotlin/io/bucketeer/sdk/android/internal/evaluation/EvaluationInteractorTest.kt

```diff
@@ -0,0 +1,185 @@
+package io.bucketeer.sdk.android.internal.evaluation
+
+import androidx.test.core.app.ApplicationProvider
+import com.google.common.truth.Truth.assertThat
+import com.squareup.moshi.Moshi
+import io.bucketeer.sdk.android.BKTConfig
+import io.bucketeer.sdk.android.internal.di.Component
+import io.bucketeer.sdk.android.internal.di.DataModule
+import io.bucketeer.sdk.android.internal.di.InteractorModule
+import io.bucketeer.sdk.android.internal.model.Evaluation
+import io.bucketeer.sdk.android.internal.model.UserEvaluationsState
+import io.bucketeer.sdk.android.internal.model.request.GetEvaluationsRequest
+import io.bucketeer.sdk.android.internal.model.response.GetEvaluationsDataResponse
+import io.bucketeer.sdk.android.internal.model.response.GetEvaluationsResponse
+import io.bucketeer.sdk.android.internal.remote.GetEvaluationsResult
+import io.bucketeer.sdk.android.mocks.evaluation1
+import io.bucketeer.sdk.android.mocks.evaluation2
+import io.bucketeer.sdk.android.mocks.user1
+import io.bucketeer.sdk.android.mocks.user1Evaluations
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import java.util.concurrent.Executors
+
+@RunWith(RobolectricTestRunner::class)
+class EvaluationInteractorTest {
+  private lateinit var server: MockWebServer
+
+  private lateinit var component: Component
+  private lateinit var moshi: Moshi
+
+  private lateinit var interactor: EvaluationInteractor
+
+  @Before
+  fun setup() {
+    server = MockWebServer()
+
+    component = Component(
+      dataModule = DataModule(
+        application = ApplicationProvider.getApplicationContext(),
+        config = BKTConfig.builder()
+          .endpoint(server.url("").toString())
+          .apiKey("api_key_value")
+          .featureTag("feature_tag_value")
+          .build(),
+      ),
+      interactorModule = InteractorModule(),
+      executor = Executors.newSingleThreadScheduledExecutor()
+    )
+
+    interactor = component.evaluationInteractor
+
+    moshi = component.dataModule.moshi
+  }
+
+  @After
+  fun tearDown() {
+    server.shutdown()
+    component.executor.shutdownNow()
+    component.dataModule.sharedPreferences.edit()
+      .clear()
+      .commit()
+  }
+
+  @Test
+  fun `fetch - initial FULL`() {
+    server.enqueue(
```

Comment:
I'm trying not to use mocks as much as possible.
So as we reach BKTClIent, tests become more like integration tests.